adapters/flipp/flipp.go

```diff
@@ -223,3 +230,38 @@ func buildBid(decision *InlineModel, impId string) *openrtb2.Bid {
 	}
 	return bid
 }
+
+func paramsUserKeyPermitted(request *openrtb2.BidRequest) bool {
+	if request.Regs != nil {
+		if request.Regs.COPPA == 1 {
+			return false
+		}
+		if request.Regs.GDPR != nil && *request.Regs.GDPR == 1 {
+			return false
+		}
+	}
+	if request.Ext != nil {
+		var extData struct {
+			TransmitEids *bool `json:"transmitEids,omitempty"`
+		}
+		if err := json.Unmarshal(request.Ext, &extData); err == nil {
+			if extData.TransmitEids != nil && !*extData.TransmitEids {
+				return false
+			}
+		}
+	}
+	if request.User != nil && request.User.Consent != "" {
+		data, err := base64.RawURLEncoding.DecodeString(request.User.Consent)
+		if err != nil {
+			return true
+		}
+		consent, err := vendorconsent.Parse(data)
+		if err != nil {
+			return true
+		}
+		if !consent.PurposeAllowed(consentconstants.ContentSelectionDeliveryReporting) {
+			return false
+		}
```

Comment:
this code path is not covered. should add  json framework test

Reply:
@onkarvhanumante I am having difficulty generating a TCF Consent string where purpose 4 is not allowed. Do you have any tips/suggestions on generating a consent string so I can add a JSON framework test for this.

Reply:
fine to skip if its difficult to generate TCF Consent string where purpose 4 is not allowed